src/app/components/BlockNavigationButtons/index.tsx

```diff
@@ -44,11 +45,13 @@ export const NextBlockButton: FC<{ scope: SearchScope; currentRound: number }> =
   return (
     <Tooltip title={disabled ? t('blocks.viewingLatest') : t('blocks.viewNext')} placement="top">
       <Box>
+        {/*If the next button is disabled, we want to poll the freshness info, because it will probably be enabled in a few secs*/}
+        {disabled && <RuntimeFreshnessTracker scope={scope} />}
```

Comment:
Home come this workaround is needed, instead of a few lines above:
```diff
- const { latestBlock } = useRuntimeFreshness(scope)
+ const { latestBlock } = useRuntimeFreshness(scope, true)
```

Reply:
i mean `const { latestBlock } = useRuntimeFreshness(scope, disabled)`

Reply:
Circular definition. Disabled is determined based on runtime freshness. So we can't use disabled as a parameter of determining runtime freshness...

Reply:
Then maybe
```ts
const staleStatus = useRuntimeFreshness(scope)
const { latestBlock } = useRuntimeFreshness(scope, currentRound >= staleStatus.latestBlock)
```

Reply:
hmmm, but staleStatus would not even be stale, if your workaround works

Reply:
> hmmm, but staleStatus would not even be stale, if your workaround works

Indeed, the data returned by the first call should be sufficient; at the second time, we just need to start the polling. So in the end I went with this:

```
  const disabled = latestBlock === currentRound
  useRuntimeFreshness(scope, disabled)
```

I propped the component which used to do the same.